src/node/structural_equal.cc

```diff
@@ -313,10 +321,38 @@ class SEqualHandlerDefault::Impl {
       *first_mismatch_ = current_paths;
     }
     if (assert_mode_ && !result) {
-      LOG(FATAL) << "ValueError: StructuralEqual check failed, caused by lhs:" << std::endl
-                 << lhs << std::endl
-                 << "and rhs:" << std::endl
-                 << rhs;
+      std::ostringstream oss;
+      oss << "ValueError: StructuralEqual check failed, caused by lhs";
```

Comment:
How do we print nodes that cannot be printed as TVMScript (e.g. Relay Nodes)?

Reply:
This is a great question. I have added the fallback mechanism to the `TVMScriptPrinter::Script` [here](https://github.com/apache/tvm/blob/0b8b8a2bab09a03258060560b487a8da5ca92f1c/src/node/script_printer.cc#L32-L34) in the latest commit. If we cannot dispatch the nodes to print (i.e. relay nodes, `Array`, `Map`, etc.), we fallback to the legacy printer.

Reply:
Could you please also comment a bit in this file, saying we will fallback to Repr printer for unsupported nodes.

Reply:
Sure, the comments have been added